src/test/kotlin/com/exactpro/epfast/template/TemplateTests.kt

```diff
@@ -0,0 +1,889 @@
+                                <tail dictionary="tail"/>
+                            </mantissa>
+                            <exponent>
+                                <delta/>
+                            </exponent>
+                        </decimal>
+                    </template>
+                """.trimIndent().byteInputStream()
+        )
+        TemplatesComparison.assertTemplateListsAreEqual(actual, expected)
+    }
+
+    @Test
+    fun `test dictionary inheritance with ascii string`() {
+        val expected = listOf(template("template") {
+            instructions {
+                asciiString("string") { copy { dictionary = "copy" } }
+            }
+        })
+
+        val actual = XmlTemplates.readFrom(
+                """
+                    <template name="template" dictionary="copy" xmlns="http://www.fixprotocol.org/ns/fast/td/1.1">
+                        <string name="string">
+                            <copy/>
+                        </string>
+                    </template>
+                """.trimIndent().byteInputStream()
+        )
+        TemplatesComparison.assertTemplateListsAreEqual(actual, expected)
+    }
+
+    @Test
+    fun `test dictionary inheritance with unicode string`() {
+        val expected = listOf(template("template") {
+            instructions {
+                unicode("string") { increment { dictionary = "increment" } }
+            }
+        })
+
+        val actual = XmlTemplates.readFrom(
+                """
+                    <templates dictionary="increment" xmlns="http://www.fixprotocol.org/ns/fast/td/1.1">
+                        <template name="template">
+                            <string charset="unicode" name="string">
+                                <increment/>
+                            </string>
+                        </template>
+                    </templates>
+                """.trimIndent().byteInputStream()
+        )
+        TemplatesComparison.assertTemplateListsAreEqual(actual, expected)
+    }
+
+    @Test
+    fun `ensure byteVector inherits dictionary`() {
+        val expected = listOf(
+            template("template") {
+                instructions {
+                    byteVector("vector") { delta { dictionary = "delta" } }
+                }
+            }
+        )
+        val actual = readTemplatesFromString(
+            """
+            <templates dictionary="dictionary" xmlns="http://www.fixprotocol.org/ns/fast/td/1.1">
+                <template name="template" dictionary="delta">
+                    <byteVector name="vector">
+                        <delta/>
+                    </byteVector>
+                </template>
+            </templates>
+            """
+        )
+        assertTemplateListsAreEqual(actual, expected)
+    }
+
+    @Test
+    fun `test dictionary inheritance with sequence`() {
+        val expected = listOf(template("template") {
+            instructions {
+                sequence(null) {
```

Comment:
NOT NOW: name should be mandatory for all instructions (excluding length/mantissa/exponent). It need to be fixed in tests and in DSL